src/item.cpp

```diff
@@ -383,12 +386,54 @@ bool item::merge_charges( const item &rhs )
     return true;
 }
 
-void item::put_in(item payload)
+// note this now returns false if unable to place item in contents[]!
+bool item::put_in(item payload)
 {
-    contents.push_back(payload);
+    if(payload.type->id == "null") {
+        debugmsg("bad value [null] for payload in item::put_in()");
+    }
+    // for now, only deal with return values and storage_used if item container
+    if(is_item_container()) {
+        if(!has_space_for(payload)) {
+            return false;
+        }
+        contents.push_back(payload);
+        storage_occupied += payload.volume();
+    } else {
+        contents.push_back(payload);
+    }
+    return true;
 }
 const char ivaresc=001;
 
+item item::take_out(unsigned int index)
+{
+    item it;
+    if(is_container_empty()) {
+        it.type->id = "null";
```

Comment:
You do not want to change the id of an existing item type, ever. This member is *only* written when new items types are created during loading of the world data.

Just return the default created item and later check the returned item with `item::is_null()`.